### PR TITLE
SAIC-725: Skip volumes with invalid statuses

### PR DESCRIPTION
**Pull request by aguzikova**

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferrylib/os/storage/cinder_database.py [hunk 1]
```diff
@@ -22,23 +23,106 @@
...
+IGNORED_TBL_LIST = ('quotas', 'quota_usages')
 QUOTA_TABLES = (
     'quotas',
     'quota_classes',
     'quota_usages',
 )
 
+# SKIPPING_STATUSES = ['creating', 'error', 'error_deleting', 'attaching',
+# 'error_attaching', 'detaching']
+VALID_STATUSES = ['available', 'in-use']
```
Comment (roman-verchikov) [inline cloudferrylib/os/storage/cinder_database.py]: We should also keep cinder volumes which are in `attaching` and `detaching` state:
 - force status change to either "available" or "in-use"
 - if changed to "available" or "in-use" - add this volume to migration procedure
 - if unable to change to "available" or "in-use" - display warning message and proceed with migration

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferrylib/os/storage/cinder_database.py [hunk 2]
```diff
@@ -22,23 +23,106 @@
...
+            table=table))
+    primary_key = cursor.fetchone().get("Column_name")
+    cursor.execute(
+        "select auto_increment from information_schema.tables"
+        " where table_name = '{table}'".format(table=table))
+    auto_increment = cursor.fetchone().get("auto_increment")
+    return primary_key, auto_increment
+
+
+def filter_data(existing_data, data_to_be_added, primary_key,
```
Comment (roman-verchikov) [inline cloudferrylib/os/storage/cinder_database.py]: Logic looks quite complex here, would be good to have it covered with unit tests

Comment (aguzikova) [inline cloudferrylib/os/storage/cinder_database.py]: The code is old. Do we need to add unit tests in this patch or in a separate one?

**Comment by archyufa** [comment]
+1 Tested in staging

**Comment by roman-verchikov** [comment]
can be merged

**Pull request merged**